internal/statemachine/classic_test.go

```diff
@@ -2393,41 +2393,23 @@ func TestBuildGadgetTreeDirectory(t *testing.T) {
 		err := stateMachine.makeTemporaryDirectories()
 		asserter.AssertErrNil(err, true)
 
-		// test the directory method
```

Comment:
hm, I think we need to re-visit the test cases of ubuntu-image. I'm not sure if this will block me from merging this PR or not, but currently I'd actually prefer this piece of code not being removed and instead look into removing the bottom portion where we're doing a `git clone`. I have no idea why William added that (and also why I missed it an only started scratching my head now). The test case is `TestBuildGadgetTreeDirectory`, so there should be no reason to have to rely on cloning a gadget from remote if we can just use a gadget that's local. I'd like to have as many unit-tests that do not depend on networking as possible (although the gadget we store locally would have to be non-network-requiring as well).

That being said, I guess we can just refactor it later on as well. Since it's a bit of a mess now anyway.

Reply:
actually this is copy-pasted, so I think it's ok to keep the other one, but this would probably have to be refactored at some point